analyses/molecular-subtyping-HGG/09-HGG-with-braf-clustering.Rmd

```diff
@@ -0,0 +1,209 @@
+  )
+)
+
+# Source the function for generating a multipanel plot
+source(
+  file.path(
+    "..",
+    "transcriptomic-dimension-reduction",
+    "util",
+    "generate-multipanel-plot-functions.R"
+  )
+)
+```
+
+
+
+### Read in data
+
+```{r}
+# Read in expression files. Note: only the relevant stranded expression score files
+# are used here as the polyA expression score files do not retain information
+# regarding low grade glioma samples.
+stranded_tsne_aligned_scores <- read_tsv(
+  file.path(data_dir,
+            "rsem_stranded_log_tsne_scores_aligned.tsv"),
+  col_types = cols(molecular_subtype = col_character())
+)
+
+stranded_umap_aligned_scores <- read_tsv(
+  file.path(data_dir,
+            "rsem_stranded_log_umap_scores_aligned.tsv"),
+  col_types = cols(molecular_subtype = col_character())
+)
+
+molecular_subtype_file <- file.path("results", "HGG_molecular_subtype.tsv")
+molecular_subtype_df <- read_tsv(molecular_subtype_file)
+```
+
+## Wrangle Data for Plotting
+
+```{r}
+# Add variables to aligned scores data.frames for plotting
+stranded_tsne_aligned_scores <- stranded_tsne_aligned_scores %>%
+  filter(broad_histology %in% c("Diffuse astrocytic and oligodendroglial tumor",
+                                "Low-grade astrocytic tumor")) %>%
+  mutate(subtype_class = case_when(
+    molecular_subtype == "BRAF V600E" &
+      broad_histology == "Diffuse astrocytic and oligodendroglial tumor" ~ "HGAT_NO_HISTONE_BRAF",
+    broad_histology == "Low-grade astrocytic tumor" ~ "LGAT",
+    TRUE ~ "OTHER_HGAT"
+  )) %>%
+  mutate(
+    HGAT_NO_HISTONE_BRAF = if_else(
+      subtype_class == "HGAT_NO_HISTONE_BRAF",
+      "yes",
+      "no"
+    )
+  )
+
+stranded_umap_aligned_scores <- stranded_umap_aligned_scores %>%
+  filter(broad_histology %in% c("Diffuse astrocytic and oligodendroglial tumor",
+                                "Low-grade astrocytic tumor")) %>%
+  mutate(subtype_class = case_when(
+    molecular_subtype == "BRAF V600E" &
+      broad_histology == "Diffuse astrocytic and oligodendroglial tumor" ~ "HGAT_NO_HISTONE_BRAF",
+    broad_histology == "Low-grade astrocytic tumor" ~ "LGAT",
+    TRUE ~ "OTHER_HGAT"
+  )) %>%
+  mutate(
+    HGAT_NO_HISTONE_BRAF = if_else(
+      subtype_class == "HGAT_NO_HISTONE_BRAF",
+      "yes",
+      "no"
+    )
+  )
+```
+
+## Dimension Reduction and Plotting
+
+```{r message = FALSE, warning = FALSE, fig.width = 20, fig.height = 8}
+# Plot generation
+stranded_tsne_plot <-
+  plot_dimension_reduction(
+    aligned_scores_df = stranded_tsne_aligned_scores,
+    point_color = "subtype_class",
+    point_size = "HGAT_NO_HISTONE_BRAF",
+    point_shape = "HGAT_NO_HISTONE_BRAF",
+    x_label = "t-SNE1",
+    y_label = "t-SNE2"
+  ) +
+  scale_color_manual(values = c("#0a7fb2", "#799d10", "#F3E500")) +
+  geom_text(aes(X1, X2, label = Kids_First_Biospecimen_ID),
+            data = filter(stranded_tsne_aligned_scores, 
+                   subtype_class == "HGAT_NO_HISTONE_BRAF"))
```

Comment:
@cbethell by conditionally adding text, we can eliminate a few steps. Specifically, if we label only the points with no histone mutation and with _BRAF_ V600E, it tells us which sample is clustering with the LGAT sample.

Reply:
Forgot to add the plot:
![HGG_stranded](https://user-images.githubusercontent.com/19534205/75043207-ac88f480-548d-11ea-9187-6791e691ec0f.png)

Reply:
Nice! 👍 This is definitely the ideal plot I pictured but in my attempt I added the `Kids_First_Biospecimen_ID` label for all the points, good to know that I can conditionally add this label in the `geom_text` function.